src/main/java/edu/kit/compiler/semantic/NamespaceMapper.java

```diff
@@ -0,0 +1,64 @@
+package edu.kit.compiler.semantic;
+
+import edu.kit.compiler.data.CompilerException;
+import edu.kit.compiler.data.ast_nodes.ClassNode;
+import edu.kit.compiler.data.ast_nodes.MethodNode;
+import lombok.Data;
+
+import java.util.HashMap;
+import java.util.Map;
+
+public class NamespaceMapper {
+
+    private final Map<Integer, ClassNamespace> symbolTableMap = new HashMap<>();
+
+    @Data
+    public static class ClassNamespace {
+        private final int classNodeRef;
+        private final Map<Integer, MethodNode.DynamicMethodNode> dynamicMethods = new HashMap<>();
+        private final Map<Integer, MethodNode.StaticMethodNode> staticMethods = new HashMap<>();
+        // Inner symbol table that holds most recent definitions of control flow
+        private final SymbolTable classSymbolTable = new SymbolTable();
+
+        public ClassNamespace(int classNodeRef) {
+            this.classNodeRef = classNodeRef;
+        }
+    }
+
+
+    /**
+     * Create a new symbol table once a new class is visited during recursive descent.
+     * @param classNode The classnode for which the new namespaces are created.
+     * @return classNamespace The class namespace which contains a reference to the class node and methods/fields
+     */
+    public ClassNamespace insertSymbolTable(ClassNode classNode) {
+        int classNodeName = classNode.getName();
+        if(symbolTableMap.containsKey(classNodeName)) {
+            // this should never happend make sure to fix all issues where the same class would get added twice
+            CompilerException.SourceLocation sl = new CompilerException.SourceLocation(classNode.getLine(), classNode.getColumn());
+            throw new SemanticException("Cannot insert same class twice into namespace map", sl);
```

Comment:
I've now had to write pretty much the same piece of code as well. This is a lot of repetition. What do you think about turning `SourceLocation` into an interface and implement it wherever we have `getLine` and `getColumn`?

Reply:
Already opened an issue yesterday. We don't have to make it an interface, but should use it instead of line/column where applicable.

Reply:
Though I would propose to open a new PR for this and close this one as fast as possible.